source/Outfit.cpp

```diff
@@ -88,6 +87,31 @@ namespace {
 				thisMap.erase(it.first);
 		}
 	}
+	
+	// Summation that avoids loss of trailing digits as much as possible.
+	double Sum(const map<double, int> &partialTerms)
+	{
+		vector<double> partialSums;
+		partialSums.reserve(partialTerms.size());
+		for(const auto &it : partialTerms)
+			partialSums.push_back(it.first * it.second);
+		sort(partialSums.begin(), partialSums.end(), [&](double a, double b){ return fabs(a) < fabs(b); });
+		return accumulate(partialSums.begin(), partialSums.end(), 0.);
+	}
+	
+	// Add value * count to an attribute.
+	// This function doesn't modify attribute if value * count is zero.
+	void AddToAttribute(double &attribute, map<double, int> &partialTerms, double value, int count)
```

Comment:
Do all additions and removals always use the proper `base value`, e.g. if we have situations where an attribute has `+0.5` and `2* -0.25`, then the magic added here wouldn't work well.

Reply:
I cannot remember it exactly, but I guess the only intention is to avoid infinitely increasing the size of "partialTerms."